src/app/collective-rewards/rewards/MyRewards.tsx

```diff
@@ -26,20 +32,91 @@ const SubText = () => {
   )
 }
 
-export const Rewards: FC<{ builder: Address }> = ({ builder }) => {
+const RewardsContent: FC<RewardDetails> = data => {
+  const { builder } = data
+
+  // We don't need to show the loading state for the backer rewards since the parent already has a loading state
+  const { data: isBacker, error: backerError } = useIsBacker(builder)
+  const { data: gauge, error: gaugeError } = useGetBuilderToGauge(builder)
+
+  const error = gaugeError ?? backerError
+
+  useHandleErrors({ error, title: 'Error loading gauge / isBacker' })
+
+  return (
+    <>
+      {gauge && gauge !== zeroAddress && (
+        <div className="pb-[46px]">
+          <BuilderRewardsSection gauge={gauge} {...data} />
+        </div>
+      )}
+      {isBacker && (
+        <BackerRewardsContextProvider backer={builder} {...data}>
+          <BackerRewardsSection {...data} />
+        </BackerRewardsContextProvider>
+      )}
+    </>
+  )
+}
+
+const BuilderRewardsSection: FC<RewardDetails & { gauge: Address }> = ({ gauge, ...data }) => {
+  const router = useRouter()
+
+  return (
+    <RewardsSection>
+      <RewardsSectionHeader
+        title="Builder Rewards"
+        subtext={<SubText />}
+        utility={<SettingsButton onClick={() => router.push('/user/settings?type=builder')} />}
+      />
+      <BuilderRewards gauge={gauge} {...data} />
+    </RewardsSection>
+  )
+}
+
+const BackerRewardsSection: FC<RewardDetails> = data => {
   const router = useRouter()
+  const canManageAllocations = useCanManageAllocations()
+  const {
+    detailedView: { value: isDetailedView, onChange: setIsDetailedView },
+  } = useBackerRewardsContext()
+
+  return (
+    <RewardsSection>
+      <RewardsSectionHeader
+        title="Backer Rewards"
+        subtext={<SubText />}
+        utility={
```

Comment:
Why do we need to define it here rather than directly in the `RewardsSectionHeader` component?

Reply:
Because the`RewardsSectionHeader ` is used for the builders and backers, and the builders does not have a Detailed View, so I had to do conditionals which does not make sense to me.